core/drivers/pm/sam/pm_suspend.S

```diff
@@ -25,10 +25,20 @@ tmp3	.req	r6
 
 /*
  * Wait until master clock is ready (after switching master clock source)
+ * @r_mckid:	register holding master clock identifier
+ *
+ * Side effects: overwrites tmp1, tmp3
  */
-.macro wait_mckrdy
+.macro wait_mckrdy r_mckid
+#ifdef CFG_SAMA7G5
+	cmp	\r_mckid, #0
+	beq	2f
+	mov	tmp3, #AT91_PMC_MCKXRDY
```

Comment:
How about using a define for the clock identifier instead of runtime tests that always will result in the same?

Reply:
Updated, thank you.